contracts/script/multisigTransactionProposals/safeSDK/upgradeProxy.ts

```diff
@@ -0,0 +1,157 @@
+import dotenv from "dotenv";
+import { ethers } from "ethers";
+import { EthersAdapter } from "@safe-global/protocol-kit";
+import SafeApiKit from "@safe-global/api-kit";
+import Safe from "@safe-global/protocol-kit";
+import { getEnvVar, createSafeTransactionData, validateEthereumAddress } from "./utils";
+const UPGRADE_PROXY_CMD = "upgradeProxy" as const;
+
+// declaring the type returned by the createTransaction method in the safe package locally (since the return type isn't exposed) so that if it's updated, it's reflected here too
+type LocalSafeTransaction = Awaited<ReturnType<Safe["createTransaction"]>>;
+
+interface UpgradeData {
+  proxyAddress: string;
+  implementationAddress: string;
+  initData: string;
+}
+
+async function main() {
+  dotenv.config();
+
+  try {
+    const upgradeData: UpgradeData = processCommandLineArguments();
+
+    /**TODO
+     * change from SEPOLIA_RPC_URL to production URL when deploying to production
+     */
+    // Initialize web3 provider using the RPC URL from environment variables
+    const web3Provider = new ethers.JsonRpcProvider(getEnvVar("SEPOLIA_RPC_URL"));
```

Comment:
nit: could just name it `RPC_URL` then we can just change the RPC_URL in .env and no code changes needed

Reply:
i like, will do

Reply:
https://github.com/EspressoSystems/espresso-sequencer/commit/391ce1ce4870422cba5ccea8303c15ec19eb554d

Reply:
Nice, added a few extra comments related to this